backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/NumaPinningHelper.java

```diff
@@ -276,6 +279,35 @@ public static void applyAutoPinningPolicy(VM vm, VdsDynamic vdsDynamic, List<Vds
         vm.setvNumaNodeList(vmNumaNodes);
     }
 
+    public static void addNumaPinningAccordingToCpuPinning(
+            VM vm,
+            Map<Integer, Collection<Integer>> cpuMappings,
```

Comment:
Do we really need the cpuMappings? The VmNumaNode should have continuous IDs, AFAIK. So each would be a range of NumOfCpus / vNUMA count (for example 16 vcpus VM with 1 vNUMA would get 0-15, if 2 vNUMA, it will get 1st vNUMA 0-7, 2nd vNUMA 8-15.
You know which physical we used based on VdsCpuUnit.